coderd/database/pubsub/pubsub.go

```diff
@@ -528,6 +555,19 @@ func (p *PGPubsub) Collect(metrics chan<- prometheus.Metric) {
 	p.qMu.Unlock()
 	metrics <- prometheus.MustNewConstMetric(currentSubscribersDesc, prometheus.GaugeValue, float64(subs))
 	metrics <- prometheus.MustNewConstMetric(currentEventsDesc, prometheus.GaugeValue, float64(events))
+
+	// additional metrics
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
```

Comment:
I worry about running a measurement with a 10s timeout from within Collect(), which typically gets called every 15s or so.

Reply:
I settled on that number because a) it's sufficiently high as to indicate a serious problem, b) it won't disrupt the default 1m scrape interval ([source](https://prometheus.io/docs/prometheus/latest/configuration/configuration/)), and c) even with a low interval like 15s it still will not interfere with scrapes.

We could lower this a bit though, what do you think?

Reply:
I guess, what's the expectation for how quickly `Collect()` returns?

```
	// This method may be called concurrently and must therefore be
	// implemented in a concurrency safe way. Blocking occurs at the expense
	// of total performance of rendering all registered metrics. Ideally,
	// Collector implementations support concurrent readers.
```

If NOTIFY is hobbled, then we're blocking `Collect()` for up to 10s, which doesn't seem like we're playing nice.

Reply:
This architecture should be fine with multiple collections, as the unique ping will just get ignored by other calls to Collect().

Reply:
OK, let's give it a try: https://github.com/coder/coder/pull/13126/commits/a7c042f5e9ccfad0afa00c7fa06f0704284bd755

Reply:
I decided to double back to synchronous-only collection: https://github.com/coder/coder/pull/13126/commits/33a2a1d2d6d2d7b50117b598ce6f8f423592a1c8
The complexity background collection introduced wasn't worth the benefit; I've satisfied my curiosity now 😅